extensions/elation/actions/addVitals/config/fields.ts

```diff
@@ -0,0 +1,246 @@
+import {
+  FieldType,
+  NumericIdSchema,
+  type Field,
+} from '@awell-health/extensions-core'
+import z, { type ZodTypeAny } from 'zod'
+
+export const fields = {
+  patientId: {
+    id: 'patientId',
+    label: 'Patient ID',
+    description: 'The patient for whom the the history is being created',
+    type: FieldType.NUMERIC,
+    required: true,
+  },
+  practiceId: {
+    id: 'practiceId',
+    label: 'Practice ID',
+    description:
+      'Practice ID where the vitas are beind measured for the patient',
+    type: FieldType.NUMERIC,
+    required: true,
+  },
+  visitNoteId: {
+    id: 'visitNoteId',
+    label: 'Visit Note ID',
+    description: 'Will only allow unsigned visit notes',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  nonVisitNoteId: {
+    id: 'nonVisitNoteId',
+    label: 'Non-Visit Note ID',
+    description: 'Will only allow unsigned non-visit notes',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  bmi: {
+    id: 'bmi',
+    label: 'BMI',
+    description:
+      'Body Mass Index; optional and calculated if height and weight are available',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  height: {
+    id: 'height',
+    label: 'Height',
+    description: 'Patient height in inches',
+    type: FieldType.STRING,
+    required: false,
+  },
+  heightNote: {
+    id: 'height note',
+    label: 'Height Note',
+    description: '',
+    type: FieldType.STRING,
+    required: false,
+  },
+  weight: {
+    id: 'weight',
+    label: 'Weight',
+    description: 'Patient weight in lbs',
+    type: FieldType.STRING,
+    required: false,
+  },
```

Comment:
The field type `string` will cause issues as data points are strongly typed in Awell. E.g. usually "weight" is collected in a form via a numeric question that results in a numeric data point value type. The builder will not be able to map this data point to this action field as the field type is string...

I'd suggest that for each measurement in Elation, we check what the logical data type would be in our system.

See https://www.loom.com/share/b0dc827e81ff4cff9488fbbbe9b83e52

Reply:
omg, I can't believe I didn't think of that. Very good point and thank you for flagging this. A valuable lesson for sure.